build/buildall

```diff
@@ -19,6 +19,8 @@ set -e
 
 shopt -s extglob
 
+git submodule update --init
```

Comment:
I'm not a fan of this.  Users don't expect a build script to change their view of the source or otherwise mess with the change control system, but that's what this does.  This needs to be removed or made an option that is off by default.